### ISSUE TITLE
Need an "add" method - which only creates a new item if another one doesn't already exist with the same key

### ISSUE THREAD

**Issue by chadkouse** [ryanfitz/vogels#32]
in DynamoDB's putItem you pass
```javascript
{
  "Expected" : {
    "someKey" : {
      "Exists": false
    }
  }
}
```

**Comment by keithchilders** [comment]
I think the support for passing ConditionExpression directly into the create() update() and destroy() methods in the 2.0 API solves this, but some object key shorthand in the options for things like item key exists and maybe attribute exists or attribute not null would be helpful with all the AttributeValue and AttributeName escaping required to write simple conditions. 

More advanced comparators like beginsWith, lt, etc. on specific attributes might even be feasible later in 2.x?

**Comment by fredrikslattman** [comment]
Like @keithchilders say, use ConditionExpression directly, here is an example:
``` javascript
var opts = {
  ConditionExpression: 'id <> :a',
  ExpressionAttributeValues: { ':a': 'my_unique_id' }
};

User.create({ id: 'my_unique_id', name: 'hello world' }, opts, function (err, user) {
  console.log(err, user);
});
```

I would however love functionality where I can specify this behavior in ```vogels.define```, that overwriting is not allowed on create, but only on put/update.

**Comment by ryanfitz** [comment]
I know this has been a long time feature request. I've been migrating to using the new Condition Expression apis. Now that that work is complete, I've added a new `overwrite` option when creating new records. When `overwrite` is set to `false` it will generate a `ConditionExpression` to prevent overwriting records with the same key.

To maintain backwards compatibility `overwrite` is defaulted to `true`. 

This feature will get released in the upcoming 2.1 release
